kubernetes/template/DeploymentTemplates/script.sh

```diff
@@ -1,263 +1,386 @@
-git clone https://github.com/msazurestackworkloads/acs-engine -b azsmaster
-cd acs-engine
+log_output -i "Installing software-properties-common and dirmngr required for Azure CLI."
+retrycmd_if_failure 5 10 sudo apt-get install software-properties-common dirmngr -y
 
-echo "We are going to use an existing ACS-Engine binary."
-echo "Open the zip file from the repo location."
-sudo mkdir bin
-sudo tar -zxvf examples/azurestack/acs-engine.tgz
-sudo mv acs-engine bin/
+log_output -i "Update system again to latest."
+retrycmd_if_failure 5 10 sudo apt-get update -y
 
-echo "Checkign if acs-engine binary is available."
-if [ -f "./bin/acs-engine" ] ; then
-	echo "Found acs-engine.exe"
-else
-	echo "Missing acs-engine.exe. Exiting!"
-	exit 1
-fi
+####################################################################################
+#Section to install Azure CLI.
+#https://docs.microsoft.com/en-us/azure/azure-stack/user/azure-stack-version-profiles-azurecli2#connect-to-azure-stack
+#https://docs.microsoft.com/en-us/cli/azure/install-azure-cli-apt?view=azure-cli-latest 
 
-EXTERNAL_FQDN="${PUBLICIP_FQDN//$PUBLICIP_DNS.$REGION_NAME.cloudapp.}"
-TENANT_ENDPOINT="https://management.$REGION_NAME.$EXTERNAL_FQDN"
+AZ_REPO=$(lsb_release -cs)
+RECV_KEY=BC528686B50D79E339D3721CEB3E94ADBE1229CF
 
-echo "EXTERNAL_FQDN is:$EXTERNAL_FQDN"
-echo "TENANT_ENDPOINT is:$TENANT_ENDPOINT"
+# Modify your sources list
+echo "deb [arch=amd64] https://packages.microsoft.com/repos/azure-cli/ $AZ_REPO main" | \
+sudo tee /etc/apt/sources.list.d/azure-cli.list
 
-retrycmd_if_failure 20 30 ensureCertificates
+#Get Microsoft signing key
+log_output -i "Get the Microsoft signing key."
+retrycmd_if_failure 5 10 sudo apt-key --keyring /etc/apt/trusted.gpg.d/Microsoft.gpg adv --keyserver packages.microsoft.com --recv-keys $RECV_KEY
 
-echo 'Retrieve the AzureStack root CA certificate thumbprint'
-THUMBPRINT=$(openssl x509 -in /var/lib/waagent/Certificates.pem -fingerprint -noout | cut -d'=' -f 2 | tr -d :)
-echo 'Thumbprint for AzureStack root CA certificate:' $THUMBPRINT
+log_output -i "Update system again to latest."
+retrycmd_if_failure 5 10 sudo apt-get update 
 
-SUFFIXES_STORAGE_ENDPOINT=$REGION_NAME.$EXTERNAL_FQDN
-SUFFIXES_KEYVAULT_DNS=.vault.$REGION_NAME.$EXTERNAL_FQDN
-FQDN_ENDPOINT_SUFFIX=cloudapp.$EXTERNAL_FQDN
-ENVIRONMENT_NAME=AzureStackCloud
+log_output -i "Installing azure cli...."
+retrycmd_if_failure 5 10 sudo apt-get install azure-cli
 
-METADATA=`curl --retry 10 $TENANT_ENDPOINT/metadata/endpoints?api-version=2015-01-01`
+log_output -i "Azure CLI version : $(az --version)"
 
-ENDPOINT_ACTIVE_DIRECTORY_RESOURCEID=`echo $METADATA  | jq '.authentication.audiences'[0] | tr -d \"`
+#####################################################################################
+#Section to install/get AKS-Engine binary.
+log_output -i "Getting AKS-Engine binary."
 
+# Todo update release branch details: msazurestackworkloads, azsmaster
```

Comment:
If all we need is a single file from the repo (`examples/azurestack/aks-engine.tgz`), then it may be faster to just download the file instead of cloning the entire repo.

`curl https://github.com/msazurestackworkloads/aks-engine/raw/azsmaster/examples/azurestack/acs-engine.tgz -o acs-engine.tgz`

Reply:
As discussed we will take above change once the full code is in AKS Upstream.